modules/drivers/vertica/src/metabase/driver/vertica.clj

```diff
@@ -134,6 +138,86 @@
        (map (partial sql.qp/->honeysql driver))
        (reduce (partial hsql/call :concat))))
 
+(defmethod sql.qp/->honeysql [:vertica :datetime-diff]
+  [driver [_ x y unit]]
+  (let [x (sql.qp/->honeysql driver x)
+        y (sql.qp/->honeysql driver y)
+        disallowed-types (keep
```

Comment:
seems wise to cast x and y to timestamp here? instead of manually casting it in every unit?

Reply:
Casting is not needed for `:day`

Reply:
any particular reason why?

Reply:
casting is needed to do `(hsql/call :<= (cast-timestamp x) (cast-timestamp y))`, since `<=` results in an error if the types of its args aren't the same. This comparison is needed if positive and negative diffs differ in implementation. day is the only unit where that's not the case.

Reply:
ah I see, thanks for the explaination